core/server/services/members/middleware.js

```diff
@@ -51,18 +51,31 @@ const exchangeTokenForSession = async function (req, res, next) {
     if (!req.url.includes('token=')) {
         return next();
     }
+
+    Object.assign(req, {action: 'signin'});
```

Comment:
Need to think up of a proper way to set the "action name". We have a concept of action-error mapper - https://github.com/TryGhost/Ghost/blob/3.2.0/core/server/web/shared/middlewares/error-handler.js#L96-L114. But here it's hard to tell what place actually defines the action? Maybe should have some kind of mapper here as well, which will detect action names/query parameters or whatever :thinking: